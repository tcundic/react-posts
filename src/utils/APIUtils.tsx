import {apiEndpoint} from "../config";
import RequestOptions from "../interfaces/Request";
import Post from "../interfaces/Post";

const request = (options: RequestOptions, additionalHeaders?: Object): Promise<any> => {
    const headers = {
        'Content-Type': 'application/json',
        ...additionalHeaders
    };

    const defaults = {headers};
    options = Object.assign({}, defaults, options, {crossDomain: true});

    return fetch(options.url, options)
        .then(response => response.json());
};

export function getPosts(): Promise<Post[]> {
    return request({
        url: `${apiEndpoint}posts`,
        method: 'GET'
    });
}

export function getPostById(id: number): Promise<Post> {
    return request({
        url: `${apiEndpoint}posts/${id}`,
        method: 'GET'
    });
}
