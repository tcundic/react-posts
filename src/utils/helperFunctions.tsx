export const greet = (greeting: string, componentName: string): void => console.log(`${greeting} ${componentName}`);