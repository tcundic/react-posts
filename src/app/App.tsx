import React, {useEffect} from 'react';
import logo from '../assets/logo.svg';
import {routes, greeting} from '../config';

import Header from '../components/Header';
import {greet} from "../utils/helperFunctions";
import Routes from "../components/Routes";

const App = () => {

    useEffect(() => greet(greeting, App.name));

    const title: string = "REACT";

    return (
        <React.StrictMode>
            <div>
                <Header greeting={greeting} logo={logo} title={title} navItems={routes}/>
                <div className="container">
                    <Routes routes={routes} greeting={greeting} />
                </div>
            </div>
        </React.StrictMode>
    );
}

export default App;
