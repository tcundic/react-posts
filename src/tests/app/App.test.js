import React from 'react';
import App from '../../app/App';
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";
import * as Greet from "../../utils/helperFunctions";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

const greeting = "Hello from";
const name = App.name;

test('should render App correctly', () => {
    const wrapper = shallow(<App/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('should print App greeting', () => {
    const greetSpy = jest.spyOn(Greet, 'greet');
    mount(<BrowserRouter><App/></BrowserRouter>);
    expect(greetSpy).toHaveBeenLastCalledWith(greeting, name);
});
