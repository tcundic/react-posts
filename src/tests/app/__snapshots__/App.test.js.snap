// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<StrictMode>
  <div>
    <Header
      greeting="Hello from"
      logo="logo.svg"
      navItems={
        Array [
          Object {
            "component": [Function],
            "isInHeader": true,
            "title": "Posts",
            "url": "/posts",
          },
          Object {
            "component": [Function],
            "isInHeader": false,
            "title": "Post",
            "url": "/post/:id",
          },
        ]
      }
      title="REACT"
    />
    <div
      className="container"
    >
      <Routes
        greeting="Hello from"
        routes={
          Array [
            Object {
              "component": [Function],
              "isInHeader": true,
              "title": "Posts",
              "url": "/posts",
            },
            Object {
              "component": [Function],
              "isInHeader": false,
              "title": "Post",
              "url": "/post/:id",
            },
          ]
        }
      />
    </div>
  </div>
</StrictMode>
`;
